spinoso-time/src/time/tzrs/parts.rs

```diff
@@ -0,0 +1,319 @@
+    #[inline]
+    #[must_use]
+    pub fn time_zone(&self) -> &str {
+        match self.offset {
+            Offset::Utc => "UTC",
+            _ => self.inner.local_time_type().time_zone_designation(),
+        }
+    }
+
+    /// Returns true if the time zone is UTC
+    ///
+    /// Can be used to implement [`Time#utc?`] and [`Time#gmt?`]
+    ///
+    //// # Examples
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let now_utc = Time::utc(2022, 7, 8, 12, 34, 56, 0);
+    /// assert!(now_utc.is_utc());
+    /// ```
+    ///
+    /// [`Time#utc?`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-utc-3F
+    /// [`Time#gmt?`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-gmt-3F
+    #[inline]
+    #[must_use]
+    pub fn is_utc(&self) -> bool {
+        Offset::Utc == self.offset
+    }
+
+    /// Returns the offset in seconds between the timezone of _time_ and UTC
+    ///
+    /// Can be used to implement [`Time#utc_offset`] and [`Time#gmt_offset`]
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let now = Time::utc(2022, 7, 8, 12, 34, 56, 0);
+    /// assert_eq!(now.utc_offset(), 0);
+    /// ```
+    ///
+    /// [`Time#utc_offset`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-utc_offset
+    /// [`Time#gmt_offset`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-gmt_offset
+    #[inline]
+    #[must_use]
+    pub fn utc_offset(&self) -> i32 {
+        self.inner.local_time_type().ut_offset()
+    }
+
+    /// Returns `true` if _time_ occurs during Daylight Saving Time in its time zone.
+    ///
+    /// Can be used to implement [`Time#dst?`] and [`Time#isdst`]
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::{Time, Offset};
+    /// use tzdb::time_zone::{europe::AMSTERDAM, pacific::AUCKLAND};
+    /// let now_ams = Time::new(2022, 5, 18, 16, 0, 0, 0, Offset::from(AMSTERDAM));
+    /// assert!(now_ams.is_dst());
+    /// let now_auckland = Time::new(2022, 5, 18, 16, 0, 0, 0, Offset::from(AUCKLAND));
+    /// assert!(!now_auckland.is_dst());
+    /// ```
+    ///
+    /// [`Time#dst?`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-dst-3F
+    /// [`Time#isdst`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-isdst
+    #[inline]
+    #[must_use]
+    pub fn is_dst(&self) -> bool {
+        self.inner.local_time_type().is_dst()
+    }
+
+    /// Returns an integer representing the day of the week, 0..6, with Sunday == 0
+    ///
+    /// Can be used to implement [`Time#wday`]
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let now = Time::utc(2022, 7, 8, 12, 34, 56, 0);
+    /// assert_eq!(now.day_of_week(), 5);
+    /// ```
+    ///
+    /// [`Time#wday`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-wday
+    #[inline]
+    #[must_use]
+    pub fn day_of_week(&self) -> u8 {
+        self.inner.week_day()
+    }
```

Comment:
I implemented all of the `is_monday`, etc, helpers because I can never remember whether 0 is Sunday or Monday-indexed (and this actually varies by locale).

If we could add those back, I'd appreciate it.